x/qgb/overview.md

```diff
@@ -0,0 +1,116 @@
+
+## [State machine](https://github.com/celestiaorg/celestia-app/tree/main/x/qgb)
+
+The state machine is the `qgb` module implementation. It is responsible for creating [attestations](https://github.com/celestiaorg/celestia-app/blob/main/x/qgb/types/attestation.go#L10-L18) which are signed by [orchestrators](https://github.com/celestiaorg/orchestrator-relayer/blob/main/docs/orchestrator.md). Each Celestia validator is expected to also run an orchestrator. [Relayers](https://github.com/celestiaorg/orchestrator-relayer/blob/main/docs/relayer.md) periodically query the Celestia state for attestations and submit them to the QGB smart contract deployed on some EVM chain.
+
+There are two types of [attestations](https://github.com/celestiaorg/celestia-app/blob/main/x/qgb/types/attestation.go#L10-L18): [valsets](https://github.com/celestiaorg/celestia-app/blob/376a1d4c0f321f12ba78279d2bd34fc6cb5e6dc2/proto/celestia/qgb/v1/types.proto#L18-L33) and [data commitments](https://github.com/celestiaorg/celestia-app/blob/376a1d4c0f321f12ba78279d2bd34fc6cb5e6dc2/proto/celestia/qgb/v1/types.proto#L35-L55).
+
+All attestations have a [`nonce`](https://github.com/celestiaorg/celestia-app/blob/8ae6a84b2c99e55625bbe99f70db1e5a985c9675/x/qgb/types/attestation.go#L16) field that defines the order in which the attestations are generated. This nonce is stored in the QGB smart contract as per [ADR-004](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-004-qgb-relayer-security.md#decision), and is used to order attestation submissions on the EVM chain.
+
+### [Valsets](https://github.com/celestiaorg/celestia-app/tree/main/x/qgb#valsets)
+
+A [valset](https://github.com/celestiaorg/celestia-app/tree/main/x/qgb#valsets) represents a validator set snapshot. It contains a list of validators' EVM addresses along with their [QGB staking power](https://github.com/celestiaorg/celestia-app/tree/main/x/qgb#validator-power-normalization). It enables the QGB smart contract to track the state of the Celestia validator set. This is necessary so that the QGB smart contract can determine when a 2/3 threshold has signed that a particular data hash has been made available on Celestia.
+
+A valset is [generated](https://github.com/celestiaorg/celestia-app/blob/34d725993a3b2c7cbbf6e62c83bbfd90ad94657e/x/qgb/abci.go#L84-L135) inside the state machine. It is then queried, signed by orchestrators, and submitted to the QGB P2P network. After more than 2/3rds of the validator set have submitted their signatures, it is relayed to the QGB smart contract along with the signatures to be [verified](https://github.com/celestiaorg/quantum-gravity-bridge/blob/3cef3f5dfd37c3086fa40a6324f144595726dc16/src/QuantumGravityBridge.sol#L172-L211) and eventually [stored](https://github.com/celestiaorg/quantum-gravity-bridge/blob/3cef3f5dfd37c3086fa40a6324f144595726dc16/src/QuantumGravityBridge.sol#L266-L268).
```

Comment:
[Question]
> A valset is [generated](https://github.com/celestiaorg/celestia-app/blob/34d725993a3b2c7cbbf6e62c83bbfd90ad94657e/x/qgb/abci.go#L84-L135) inside the state machine.

What triggers the generation of an attestation? it may also make sense to mention it in the text, up to you.

> It is then queried, signed by orchestrators, and submitted to the QGB P2P network.

It would be great if the actors could be specified in this sentence i.e., who performs the query (relayer perhaps)?  and who submits the attestation to the GGB p2p netowrk? Also, some clarification on what constitutes QGB P2P network would be great (is it the rollup chain? Ethereum p2p layer? etc)

Reply:
> What triggers the generation of an attestation?

The link in `generated`  shows where it is generated inside the state machine

> It is then queried, signed by orchestrators, and submitted to the QGB P2P network.

All of these actions are done by the orchestrator.

> some clarification on what constitutes QGB P2P network would be great (is it the rollup chain? Ethereum p2p layer? etc)

The QGB P2P network is a separate P2P network. 

[d00699e](https://github.com/celestiaorg/celestia-app/pull/1974/commits/d00699ec1de18c8c95899d785e72a6b37da9b228) This should clarify this

Reply:
> The link in generated shows where it is generated inside the state machine

It would be really useful to give a brief overview of what causes the generation of an attestation, like a specific SDK message. This way, specifications are self-contained and do not necessitate readers to review the code.